api/src/main/java/marquez/api/DatasetResource.java

```diff
@@ -238,6 +245,12 @@ void throwIfNotExists(@NonNull NamespaceName namespaceName, @NonNull DatasetName
     }
   }
 
+  void throwIfSourceNotExists(SourceName sourceName) throws MarquezServiceException {
+    if (!sourceDao.exists(sourceName.getValue())) {
```

Comment:
We'll want to use [`SourceService.exists()`](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/service/SourceService.java#L69) that way the resource layer only depends on the service layer.